frontend/src/games/sequence-game.tsx

```diff
@@ -0,0 +1,89 @@
+
+import { useEffect, useRef, useState } from 'react';
+
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const SequenceGame = () => {
+  const [sequence, setSequence] = useState<(number)[]>([]);
+  const userSequence = useRef<number[]>([]);
+  const [isGameStarted, setIsGameStarted] = useState(false);
+  const [litUpButtonId, setLitUpButtonId] = useState<number>();
+
+  useEffect(() => {
+    if (sequence.length > 0) {
+      showSequence().then(() => setLitUpButtonId(undefined));
+    }
+  }, [sequence]);
+  
+  const showSequence = async () => {
+    for (const buttonId of sequence) {
+      setLitUpButtonId(buttonId);
+      await delay(200);
+      setLitUpButtonId(undefined);
+      await delay(100);
+    }
+  }
+  
+  const getSequence = async () => {
+    try {
+      const querySuffix = sequence.length != 0 ? `?sequence=${sequence.toString()}` : ``;
+      const response = await fetch(`http://localhost:5252/SequenceGame/getSequence${querySuffix}`);
+      const newSequence = await response.json();
+      
+      setSequence(newSequence);
+      setIsGameStarted(true);
+    } catch (error) {
+      console.error("Error starting game: ", error);
+    }
+  }
+  
+  const handleClick = (buttonId: number) => {
+    if (!isGameStarted) return;
+    
+    setLitUpButtonId(buttonId);
+    delay(100).then(() => setLitUpButtonId(undefined));
+    
+    userSequence.current.push(buttonId);
```

Comment:
Not sure ar geriausia ideja cia naudoti masyva, speju labiau efficient butu tiesiog tureti koki kintamaji prie kurio tsg visada +1 pridedi, bet veikia ir taip tai manau neverta keisti nieko, tas performance difference butu ziauriai ziauriai ziauriai mazas